### PR TITLE
step 4 - core:get /api/articles/:article_id

### PR DESCRIPTION
**Pull request by mikef80**

### CODE REVIEW

File: __tests__/articles.test.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article.article_id).toBe(1);
+        expect(typeof body.article).toBe("object");
+      });
+  });
+
+  it("GET:200 response object has the following properties: author, title, article_id, body, topic, created_at, votes, article_img_url", () => {
```
Comment (nc-jonathan-mcguire) [inline __tests__/articles.test.js:26]: I would maybe just say `has the correct properties` - we can then look at the test to see what these are

File: __tests__/articles.test.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toHaveProperty("author");
+        expect(body.article).toHaveProperty("title");
+        expect(body.article).toHaveProperty("article_id");
+        expect(body.article).toHaveProperty("body");
+        expect(body.article).toHaveProperty("topic");
+        expect(body.article).toHaveProperty("created_at");
+        expect(body.article).toHaveProperty("votes");
+        expect(body.article).toHaveProperty("article_img_url");
```
Comment (nc-jonathan-mcguire) [inline __tests__/articles.test.js:38]: You'd ideally want to check these are the correct types as well. 

Crutially, we'd want to make sure we've got back article_id of 1, as thats the one you've requested. At the moment, this matches for all

File: __tests__/articles.test.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  it("GET:400 responds with 'bad request' if article if provided with an invalid article_id", () => {
+    return request(app)
+      .get("/api/articles/chickens")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
+  });
+});
```
Comment (nc-jonathan-mcguire) [inline __tests__/articles.test.js:59]: nice sad paths

File: api/errors.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+exports.handlePSQLErrors = (err, req, res, next) => {
+  if ((err.code = "22P02")) {
+    res.status(400).send({ msg: "bad request" });
+  } else {
+    next(err);
+  }
+};
+
+exports.handleServerErrors = (err, req, res, next) => {
```
Comment (nc-jonathan-mcguire) [inline api/errors.js:17]: nice error handlers

**Review by nc-jonathan-mcguire — CHANGES_REQUESTED** [review]
Some nice stuff here, just needs a change to a test:

You'd ideally want to check these are the correct types as well. 

Crutially, we'd want to make sure we've got back article_id of 1, as thats the one you've requested. At the moment, this matches for all

Can use toMatchObject for this

File: __tests__/articles.test.js [hunk 5]
```diff
@@ -43,6 +36,14 @@ describe("/api/articles/:article_id", () => {
         expect(body.article).toHaveProperty("created_at");
         expect(body.article).toHaveProperty("votes");
         expect(body.article).toHaveProperty("article_img_url");
+        expect(typeof body.article.author).toBe("string");
+        expect(typeof body.article.title).toBe("string");
+        expect(typeof body.article.article_id).toBe("number");
+        expect(typeof body.article.body).toBe("string");
+        expect(typeof body.article.topic).toBe("string");
+        expect(new Date(body.article.created_at)).toBeInstanceOf(Date);
```
Comment (Hard-Luck) [inline __tests__/articles.test.js:44]: because you're invoking this with the new keyword it will always be an instance of Date

**Review by Hard-Luck — COMMENTED** [review]
Good work, feel free to merge

File: __tests__/articles.test.js (hunk 5 above)
Comment (mikef80) [reply __tests__/articles.test.js:44]: @Hard-Luck  Thank you!  I wasn't sure if the Date constructor would throw an error if passed something that couldn't be turned into a date.  Would there be a better way to deal with this check?

**Pull request merged**